text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+<!-- tocstop -->
+
+## Motivation
+
+This proposal attempts to focus on the following problems within OpenTelemetry to unblock multiple working groups:
+
+- Allowing mutating attributes to participate in Resource ([OTEP 208](https://github.com/open-telemetry/oteps/pull/208)).
+- Allow Resource to handle entities whose lifetimes don't match the SDK's lifetime ([OTEP 208](https://github.com/open-telemetry/oteps/pull/208)).
+- Provide support for async resource lookup ([spec#952](https://github.com/open-telemetry/opentelemetry-specification/issues/952)).
+- Fix current Resource merge rules in the specification, which most implementations violate ([oteps#208](https://github.com/open-telemetry/oteps/pull/208), [spec#3382](https://github.com/open-telemetry/opentelemetry-specification/issues/3382), [spec#3710](https://github.com/open-telemetry/opentelemetry-specification/issues/3710)).
+- Allow semantic convention resource modeling to progress ([spec#605](https://github.com/open-telemetry/opentelemetry-specification/issues/605), [spec#559](https://github.com/open-telemetry/opentelemetry-specification/issues/559), etc).
+
+## Design
+
+### Approach - Resource Improvements
+
+Let's focus on outlining Entity detectors and Resource composition.  This has a higher priority for fixing within OpenTelemetry, and needs to be unblocked sooner. Then infer our way back to data model and Collector use cases.
+
+We define the following SDK components:
+
+- **Resource Detectors (legacy)**:  We preserve existing resource detectors.  They have the same behavior and interfaces as today.
+- **Entity Detectors (new)**: Detecting an entity that is relevant to the current instance of the SDK.  For example, this would detect a service entity for the current SDK, or its process. Every entity must have some relation to the current SDK.
+- **Resource Coordinator (new)**: A component responsible for taking Resource and Entity detectors and doing the following:
+  - Constructing a Resource for the SDK from detectors.
+  - Dealing with conflicts between detectors.
+  - Providing SDK-internal access to detected Resources for reporting via Log signal on configured LogProviders.
+  - *(new) Managing Entity changes during SDK lifetime, specifically dealing with entities that have lifetimes shorter than the SDK*
+
+#### Resource Coordinator
+
+The SDK Resource coordinator is responsible for running all configured Resource and Entity Detectors.  There will be some (user-controlled, otel default) priority order to these.
+
+- The resource coordinator will detect conflicts in Entity of the same type being discovered and choose one to use.
+- When using Entity Detectors and Resource detectors together, the following merge rules will be used:
+  - Entity merging will occur first resulting in an "Entity Merged" Resource.
+    - Entities of different types will be merged into the resulting Resource.
+    - Entities of the same type will have one rejected and one accepted, based on priority.
```

Comment:
Yes - I think we can merge if the identities are the same, and add descriptive attributes.  I can update this appropriately, or if you have a proposal I'll add it directly here.

Reply:
Let me flesh out this section more given comments, but a quick answer to identity:

Since Entity can have descriptive attributes, when two entities have the same type, we can merge descriptive attributes.

Regarding schema_url -> Nominally when merging two of the "same" entities, we'd need to execute schema_url translations to do so, e.g. see: https://github.com/open-telemetry/oteps/pull/161

So you're right about identity, but also I think we have two "identities" to think about here:

- When are two entities describing the same thing and can be trivially merged
- When are two entities describing the same thing and can be merged by interacting with schema_url

I'm not sure how to talk about this "simply" yet, so I'll work on expanding the merge algorithm to show case this, and be more explicit about what fields are used for "identity" in each case.

Reply:
>Regarding schema_url -> Nominally when merging two of the "same" entities, we'd need to execute schema_url translations to do so, e.g. see

+1.